examples/performance-debugging/performance-debugging.cpp

```diff
@@ -462,16 +461,16 @@ int main(int argc, char* argv[])
         // Similarly time the apply
         exec->synchronize();
         auto a_tic = std::chrono::steady_clock::now();
-        generated_solver->apply(gko::lend(b), gko::lend(x_clone));
+        generated_solver->apply(b, x_clone);
         exec->synchronize();
         auto a_tac = std::chrono::steady_clock::now();
         auto apply_time =
             std::chrono::duration_cast<std::chrono::nanoseconds>(a_tac - a_tic);
         output_file << "Apply time (ns): " << apply_time.count() << std::endl;
 
         // Compute the residual norm
-        auto residual = utils::compute_residual_norm(gko::lend(A), gko::lend(b),
-                                                     gko::lend(x_clone));
+        auto residual =
+            utils::compute_residual_norm(A.get(), b.get(), x_clone.get());
```

Comment:
Any reason why you did not implement these `utils` functions with `pointer_param`?
Would the `ValueType` template parameter not get automatically deduced?

Reply:
yes, template argument deduction needs an exact match, so conversions like the one to `pointer_param` can't involve template parameters.